### DISCUSSION TITLE
License question

### DISCUSSION THREAD

**Discussion by HSete** [winsiderss/systeminformer#1055] [Questions / Answers] [answered]
I'm adding Process Hacker to a useful software package and as usual I compile all program licenses into a single big TXT file... So I would like to know which files need to be in my big license file, COPYRIGHT.txt, LICENSE.txt or both?

Thanks in advance.

**Comment by Masamune3210** [comment]
From a quick glance those files seem to be copies of each other, mostly just including the GNU license statement

**Reply by HSete** [reply]
What confused me is that the COPYRIGHT.txt file mentions the authors, BUT the LICENSE.txt file, in addition to being mentioned in the COPYRIGHT.txt file, has "exceptions" that are not in the COPYRIGHT.txt file

**COPYRIGHT.txt**
```
== Process Hacker ==
Process Hacker is licensed under the GNU GPL v3, with exceptions. A full
copy of the license is provided in LICENSE.txt.

    Copyright (C) 2009-2016 wj32 and various authors

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.

== Mini-XML ==
Process Hacker uses Mini-XML licensed under the following terms:
...
```

**LICENSE.txt**

```
Process Hacker is distributed under the GNU GPL version 3, with the
following exception:

    Permission is granted to dynamically (but not statically) link this
    program with independent modules, regardless of the license terms of
    these independent modules, provided that this program is not modified
    in any way. An independent module is a module which is not derived
    from or based on this program. If you modify this program, this
    additional permission no longer applies unless authorized by the
    copyright holders.

                    GNU GENERAL PUBLIC LICENSE
                       Version 3, 29 June 2007
...
```

**Reply by Masamune3210** [reply]
As it states at the top of COPYRIGHT that LICENSE is full copy of the license, I'm going to probably assume that would be the one to use. If you wanted to be safe you could take the differing excerpts between the two and add them to the final file to make sure you catch everything? Or possibly provide a link to this repo so people could inquire upon the files themselves if they want to

**Comment by dmex** [comment]
> As it states at the top of COPYRIGHT that LICENSE is full copy of the license, I'm going to probably assume that would be the one to use.

LICENSE has the GPLv3 license but not the project license which is GPLv3 with that exemption for plugins/dynamic linking.

A) If you're using our sources in-house and not sharing binaries with the public then you don't need to share changes or sourcecode.
B) If you are distributing to the public then the licenses for those other components and the project itself all basically require you make the source code available for download also reference this project somewhere like an readme.txt or an about window - You can use TaskExplorer as an example since it was created and built using our sources: https://github.com/DavidXanatos/TaskExplorer

> If you wanted to be safe you could take the differing excerpts between the two and add them to the final file to make sure you catch everything? 

COPYRIGHT.txt hasn't been updated in a few years and doesn't include a other components added since then. There would need to be additions for each license file in the directories here: https://github.com/processhacker/processhacker/tree/master/tools/thirdparty

I should probably create an issue for copyright.txt to get updated?

**Reply by HSete** [reply]
It would be great if the license file were updated.

==

If I understood correctly:
Distributing the binaries (even if not modified) should I put in my license file a reference to Process Hacker (GPLv3) and also a reference to each plugin that is in the pointed directory? Correct?

**Reply by ionescu007** [reply]
@dmex @jxy-s We should post an answer here now with the MIT license and cleanups that we've done.

**Comment by jxy-s** [comment, marked as answer]
The project is now licensed under the MIT license. Trademarks and copyrights associated with the project were transferred to Winsider Seminars & Solutions, Inc. on June 1st 2022.

We have updated the project license and files to reflect these changes. Please see the source files for the full information.

**Comment by atomizer** [comment]
Is it really so easy to remove GPL?

**Reply by Masamune3210** [reply]
No, one of the points of the GPL is that you can't really resend it once it is given. Any code that is shared must include the license text in its entirety. It doesn't matter to most people since most people won't directly be interacting with the code, and it's not a license agreement so it doesn't affect usage of the compiled program outside of needing to provide a way to acquire the source of the program to the user if they want it

**Comment by ionescu007** [comment]
The GPL license has not been rescinded or removed. All code currently under
GPL as part of Process Hacker continues to be licensed as such.

What a copyright author can choose to do, however, is to provide code and
re-license it under a different license. This doesn't invalidate the
previous license, and the old code must continue to be provided under the
old license.

So System Informer -- is a re-licensed version of Process Hacker under a
more permissive, and GPL-compatible license. Someone theoretically is free
to take every single check-in done to System Informer and "port it" to the
GPL-ed Process Hacker source, and release that.

Best regards,
Alex Ionescu
